wordsquiz/src/components/common/quiz/QuizSheet.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect } from "react";
+import { Grid, Container, Typography, Button } from "@mui/material";
+import { addDoc, collection } from "firebase/firestore";
+import { useFirebase } from "../../contexts/FirebaseContext";
+import CustomTextField from "../CustomTextField";
+import StopIcon from "@mui/icons-material/Stop";
+
+export default function QuizSheet({
+  wordsToQuiz,
+  setQuizActive,
+  setQuizFinished,
+  setQuizSheetData,
+}) {
+  const [quizSheet, setQuizSheet] = useState([]);
+  const [answer, setAnswer] = useState(""); // ten stan wg mnie jest niepotrzebny - do potwierdzenia.
+  const firebase = useFirebase();
+
+  useEffect(() => {
+    let allWords = [];
+    wordsToQuiz.forEach(
```

Comment:
Byłoby lepiej użyć funkcji reduce do zbudowania tej tablicy. Poza tym uzywaj const wszedzie gdzie to mozliwe.

Reply:
ok, przerobię.

Reply:
Odnośnie consta - nie mogę go użyć, bo potem przypisuję do tej zmiennej inną wartość, a jak zrobię z niej consta to nie mogę. Musiałbym chyba utworzyć do niej stan i ustawiać go setterem, co wydaje mi się bez sensu na potrzeby tej metody.
No chyba, że inaczej jakoś trzeba do tego podejść?